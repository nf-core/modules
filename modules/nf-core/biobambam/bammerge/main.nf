process BIOBAMBAM_BAMMERGE {
    tag "$meta.id"
    label 'process_low'

    conda (params.enable_conda ? "bioconda::biobambam=2.0.183" : null)
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/biobambam:2.0.183--h9f5acd7_1':
        'quay.io/biocontainers/biobambam:2.0.183--h9f5acd7_1' }"

    input:
    tuple val(meta), path(bam)

    output:
    tuple val(meta), path("${prefix}.bam")  ,emit: bam
    tuple val(meta), path("*.bai")          ,optional:true, emit: bam_index
    tuple val(meta), path("*.md5")          ,optional:true, emit: checksum
    path "versions.yml"                     ,emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    prefix = task.ext.prefix ?: "${meta.id}"
    def input_string = bam.join(" I=")

    """
    bammerge \\
        I=${input_string} \\
        $args \\
        > ${prefix}.bam

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        bammerge: \$( bammerge --version |& sed '1!d; s/.*version //; s/.\$//' )
    END_VERSIONS
    """
}
