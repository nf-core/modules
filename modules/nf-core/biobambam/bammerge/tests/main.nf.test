
nextflow_process {

    name "Test Process BIOBAMBAM_BAMMERGE"
    script "../main.nf"
    process "BIOBAMBAM_BAMMERGE"
    config "./nextflow.config"

    tag "modules"
    tag "modules_nfcore"
    tag "biobambam"
    tag "biobambam/bammerge"

    test("test-biobambam-bammerge-paired") {

        when {
            process {
                """
                input[0] = [
				    [ id:'test', single_end:false ], // meta map
				    [
				        file(params.modules_testdata_base_path + 'genomics/homo_sapiens/illumina/bam/test.paired_end.sorted.bam', checkIfExists: true),
				        file(params.modules_testdata_base_path + 'genomics/homo_sapiens/illumina/bam/test2.paired_end.sorted.bam', checkIfExists: true)
				    ]
                ]

                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(
					bam(process.out.bam[0][1]).getReadsMD5(),
					file(process.out.bam_index[0][1]).name,
					process.out.checksum,
					process.out.versions
					).match()
				}
            )
        }
    }

    test("test-biobambam-bammerge-single") {

        when {
            process {
                """
                input[0] = [
				    [ id:'test', single_end:false ], // meta map
				    [
				        file(params.modules_testdata_base_path + 'genomics/sarscov2/illumina/bam/test.single_end.sorted.bam', checkIfExists: true),
				    ]
                ]

                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(
					bam(process.out.bam[0][1]).getReadsMD5(),
					file(process.out.bam_index[0][1]).name,
					process.out.checksum,
					process.out.versions
					).match()
				}
            )
        }
    }

}
