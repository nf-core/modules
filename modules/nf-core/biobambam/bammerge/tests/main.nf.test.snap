{
    "test-biobambam-bammerge-paired": {
        "content": [
            "b21a14064887edf38c798893228c30ba",
            "test.bam.bai",
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "test.md5:md5,31c59857990ceb392242136429e30243"
                ]
            ],
            [
                "versions.yml:md5,9a4e6cfa364f1f6ef9112fa248d66fea"
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-08-28T10:28:49.100225"
    },
    "test-biobambam-bammerge-single": {
        "content": [
            "94fcf617f5b994584c4e8d4044e16b4f",
            "test.bam.bai",
            [
                [
                    {
                        "id": "test",
                        "single_end": false
                    },
                    "test.md5:md5,244a9d1cbc6d74724285c80220e5e427"
                ]
            ],
            [
                "versions.yml:md5,9a4e6cfa364f1f6ef9112fa248d66fea"
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-08-28T10:27:48.721949"
    }
}